core/models/user.py

```diff
@@ -272,17 +273,30 @@ def health_facility(self):
 
     @property
     def is_officer(self):
-        return Officer.objects.filter(
-            code=self.username, has_login=True, validity_to__isnull=True).exists()
+        cache_name = f"user_eo_{self.username}"
+        is_officer = cache.get(cache_name)
+        if not is_officer:
```

Comment:
`if is_officer is None` instead? Otherwise it will cause a cache miss when the cached value is `False`